src/doc/src/reference/profiles.md

```diff
@@ -0,0 +1,394 @@
+the `--debug` flag to switch to the `dev` profile.
+
+Test targets are built with the `test` profile by default. The `--release`
+flag switches tests to the `bench` profile.
+
+Bench targets are built with the `bench` profile by default. The [`cargo
+build`] command can be used to build a bench target with the `test` profile to
+enable debugging.
+
+Note that when using the [`cargo test`] and [`cargo bench`] commands, the
+`test`/`bench` profiles only apply to the final test executable. Dependencies
+will continue to use the `dev`/`release` profiles. Also note that when a
+library is built for unit tests, then the library is built with the `test`
+profile. However, when building an integration test target, the library target
+is built with the `dev` profile and linked into the integration test
+executable.
+
+[`cargo bench`]: ../commands/cargo-bench.md
+[`cargo build`]: ../commands/cargo-build.md
+[`cargo check`]: ../commands/cargo-check.md
+[`cargo install`]: ../commands/cargo-install.md
+[`cargo run`]: ../commands/cargo-run.md
+[`cargo rustc`]: ../commands/cargo-rustc.md
+[`cargo test`]: ../commands/cargo-test.md
+
+### Overrides
+
+Profile settings can be overridden for specific packages and build-time
+crates. To override the settings for a specific package, use the `package`
+table to change the settings for the named package:
+
+```toml
+# The `foo` package will use the -Copt-level=3 flag.
+[profile.dev.package.foo]
+opt-level = 3
+```
+
+The package name is actually a [Package ID Spec](pkgid-spec.md), so you can
+target individual versions of a package with syntax such as
+`[profile.dev.package."foo:2.1.0"]`.
+
+To override the settings for all dependencies (but not any workspace member),
+use the `"*"` package name:
+
+```toml
+# Set the default for dependencies.
+[profile.dev.package."*"]
+opt-level = 2
+```
+
+To override the settings for build scripts, proc macros, and their
+dependencies, use the `build-override` table:
+
+```toml
+# Set the settings for build scripts and proc-macros.
+[profile.dev.build-override]
+opt-level = 3
+```
+
+> Note: When a dependency is both a normal dependency and a build dependency,
+> Cargo will try to only build it once when `--target` is not specified. When
+> using `build-override`, the dependency may need to be built twice, once as a
+> normal dependency and once with the overridden build settings. This may
+> increase initial build times.
+
+The precedence for which value is used is done in the following order (first
+match wins):
+
+1. `[profile.dev.package.name]` — A named package.
+2. `[profile.dev.package."*"]` — For any non-workspace member.
+3. `[profile.dev.build-override]` — Only for build scripts, proc macros, and
+   their dependencies.
+4. `[profile.dev]` — Settings in `Cargo.toml`.
+5. Default values built-in to Cargo.
+
+Overrides cannot specify the `panic`, `lto`, or `rpath` settings.
+
+#### Overrides and generics
```

Comment:
@alexcrichton Just wanted to check if you had a chance to read this section. I wasn't confident it is correct or clear or goes into too much or too little detail. Particularly the last paragraph.

Reply:
Heh good catch!

Looks accurate to me though. The only thing I might recommend is a "if you've got questions and/or are confused about this, please let us know" style comment. Basically I just want to make sure folks know they can contact us if they're weirded out.